src/shims/foreign_items.rs

```diff
@@ -167,6 +170,30 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
         };
 
         // Next: functions that return.
+        match link_name {
+            "__rust_maybe_catch_panic" => {
```

Comment:
Why'd you move this up?

Reply:
I did this provisionally because of the early return. I haven't decided how to solve it but maybe emulate_foreign_item_by_name` should return a `InterpResult<bool>` instead and use the boolean value to know if its necessary to return earlier.